<!DOCTYPE html>
<html>
<head lang="en">
    <meta charset="UTF-8">
    <title></title>
</head>
<body>
    <script>
        window.onerror = function(msg,url,line){
             print('Message : ' + msg);
             print('url: ' + url);
             print('line: ' + line);
        }
        function print(str){
            document.write(str+'<br>');
        }
    </script>
    <script>
        try{
             throw {a:'test', b:false};
        } catch(e){
             print(e);
        } finally {
            print('In finally clause');
        }
    </script>
    <script>
        function myFunc(){
            document.write1('Value of variable a is: ' + a);
            try{
                var a = 100;
                document.write1('Value of variable a is: ' + a);
            } catch(e){
                print(e.message);
                print(e.stack);
            } finally {
                throw 'new exception!';
            }


        }
    </script>
    <form>
        <input type="button" value="Click Me!" onclick="myFunc();"/>
    </form>
</body>
</html>